core/common/src/main/kotlin/com/unifest/android/core/common/utils/ImageLuminanceUtils.kt

```diff
@@ -0,0 +1,130 @@
+                    .data(imageUrl)
+                    .allowHardware(false)
+                    .build()
+
+                val result = imageLoader.execute(request)
+
+                Timber.d("ImageLoader result: ${result::class.simpleName}")
+                if (result is SuccessResult) {
+                    val drawable = result.drawable
+                    Timber.d("Drawable type: ${drawable::class.simpleName}")
+
+                    val bitmap = when (drawable) {
+                        is BitmapDrawable -> {
+                            val originalBitmap = drawable.bitmap
+                            // HARDWARE bitmap은 pixel 접근이 불가능하므로 SOFTWARE로 복사
+                            if (originalBitmap.config == Bitmap.Config.HARDWARE) {
+                                originalBitmap.copy(Bitmap.Config.ARGB_8888, false)
+                            } else {
+                                originalBitmap
+                            }
+                        }
+
+                        else -> {
+                            Timber.d("Drawable is not BitmapDrawable, returning null")
+                            return@withContext null
+                        }
+                    }
+
+                    val luminanceResult = calculateBitmapLuminance(bitmap)
+                    Timber.d("Luminance calculation complete: $luminanceResult")
+                    luminanceResult
+                } else {
+                    Timber.d("ImageLoader request failed: $result")
+                    null
+                }
+            } catch (e: Exception) {
+                Timber.d("Exception during luminance calculation: ${e.message}")
+                e.printStackTrace()
+                null
+            }
+        }
+    }
+
+    private fun calculateBitmapLuminance(bitmap: Bitmap): LuminanceResult {
+        val width = bitmap.width
+        val height = bitmap.height
+        var totalLuminance = 0.0
+
+        // 상단 30% 영역만 샘플링 (status bar와 top app bar 영역)
+        val sampleHeight = (height * 0.3).toInt()
+        // 성능을 위한 샘플링 간격(모든 픽셀을 다 검사하지 않고 일정 간격으로만 픽셀을 샘플링해서 계산 속도를 높임)
+        val stepX = maxOf(1, width / 50)
+        val stepY = maxOf(1, sampleHeight / 20)
+        var sampleCount = 0
+
+        for (y in 0 until sampleHeight step stepY) {
+            for (x in 0 until width step stepX) {
+                val pixel = bitmap[x, y]
+                val red = (pixel shr 16) and 0xFF
+                val green = (pixel shr 8) and 0xFF
+                val blue = pixel and 0xFF
+
+                // ITU-R BT.709 luminance formula
+                // 0.2126 (Red): 인간의 눈이 빨간색에 가장 덜 민감
+                // 0.7152 (Green): 인간의 눈이 초록색에 가장 민감 (약 71%)
+                // 0.0722 (Blue): 인간의 눈이 파란색에 두 번째로 덜 민감
+                val luminance = (0.2126 * red + 0.7152 * green + 0.0722 * blue) / 255.0
+                totalLuminance += luminance
+                sampleCount++
+            }
+        }
```

Comment:
사소한 차이가 예상되지만,
비트맵 연산에서 꽤 많은 CPU 연산량이 요구될 것 같은데 `withContext(Dispatchers.Default)` 로 연산을 수행하는 건 어떨까요 ??

```suggestion
private suspend fun calculateBitmapLuminance(bitmap: Bitmap): LuminanceResult =
        withContext(Dispatchers.Default) { 
        // ... 
        LuminanceResult(averageLuminance, isDark)
}

```

Reply:
그렇네요 말씀하신것 처럼 Default가 더 적합한 것 같습니다. 바꿀께요!